static/js/metrics-explorer.js

```diff
@@ -393,25 +397,65 @@ function formulaInputHandler(formulaElement, uniqueId) {
         }, 500)
     ); // debounce delay
 }
+function extractFunctionsAndFormula(formulaInput) {
+    const parseObject = {
+        formula: '',
+        functions: [],
+    };
+
+    // Define a regular expression to match functions
+    const functionPattern = /\b(\w+)\s*\(([^()]*)\)/g;
+    let match;
+    const functionsFound = [];
+
+    // Capture functions in the order they appear
+    while ((match = functionPattern.exec(formulaInput)) !== null) {
+        functionsFound.push(match[1]);
+        // Replace the matched function with its content for further processing
+        formulaInput = formulaInput.replace(match[0], match[2]);
+        functionPattern.lastIndex = 0; // Reset the regex index after replacement
+    }
+
+    // Reverse to maintain the correct order of function execution
+    parseObject.functions = functionsFound;
+
+    // The remaining part of the formulaInput should be the innermost formula
+    parseObject.formula = formulaInput.trim();
+
+    return parseObject;
+}
+function appendFormulaFunctionAlertDiv(formulaElement, fnNames) {
+    if (!Array.isArray(fnNames)) {
+        throw new TypeError('fnNames should be an array');
+    }
+
+    fnNames.forEach((fnName) => {
+        var newDiv = $('<div class="selected-function-formula">' + fnName + '<span class="close">×</span></div>');
+        formulaElement.find('.all-selected-functions-formula').append(newDiv);
+    });
+}
 
 async function addAlertsFormulaElement(formulaInput) {
     let uniqueId = generateUniqueId();
     let queryNames = Object.keys(queries);
     if (!formulaInput) {
         formulaInput = queryNames.join(' + ');
     }
+    let formulaandfunc = extractFunctionsAndFormula(formulaInput);
```

Comment:
Looks good. 
nit : rename the variable `formulaandfunc` to `formulaAndFunctions`